funtofem/optimization/utils/plot_manager.py

```diff
@@ -0,0 +1,188 @@
+        elif omag == 3:
+            return r"*$10^3$"
+        elif omag == 4:
+            return r"*$10^4$"
+        elif omag == 5:
+            return r"*$10^5$"
+        elif omag == 6:
+            return r"*$10^6$"
+        elif omag == -1:
+            return r"*$10^{-1}$"
+        elif omag == -2:
+            return r"*$10^{-2}$"
+        elif omag == -3:
+            return r"*$10^{-3}$"
+        elif omag == -4:
+            return r"*$10^{-4}$"
+        elif omag == -5:
+            return r"*$10^{-5}$"
+        elif omag == -6:
+            return r"*$10^{-6}$"
+
+    @property
+    def iterations(self) -> list:
+        return [_ for _ in range(len(self.hist_dict_list))]
+
+    def __call__(
+        self,
+        plot_name=None,
+        show_scales=False,
+        legend_frac: float = 0.8,
+        yaxis_name="func values",
+        color_offset: int = 0,
+    ):
+        if plot_name is None:
+            plot_name = "f2f-history"
+        plt.style.use(niceplots.get_style())
+        fig, ax = plt.subplots()
+        colors = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+        for ifunc, func in enumerate(self.functions):
+            if self.valid_function(func):
+                if show_scales:
+                    omag = int(np.floor(np.log(func.scale) / np.log(10)))
+                    label = func.plot_name + self.get_r_string(omag)
+                else:
+                    label = func.plot_name
+                ax.plot(
+                    self.iterations,
+                    self.get_hist_array(func),
+                    linewidth=2,
+                    color=colors[ifunc + color_offset],
+                    label=label,
+                )
+            else:
+                print(f"Warning function {func.name} is not in function history list..")
+        grey_colors = plt.cm.Greys(np.linspace(1.0, 0.5, len(self.constr_dicts)))
+        for icon, constr_dict in enumerate(self.constr_dicts):
+            name = constr_dict["name"]
+            value = constr_dict["value"]
+            ax.plot(
+                self.iterations,
+                np.array([value for _ in self.iterations]),
+                linestyle="dotted",
+                linewidth=2,
+                color=grey_colors[icon],
+                label=name,
+            )
+        # put axis on rhs of plot
+        box = ax.get_position()
+        ax.set_position([box.x0, box.y0, box.width * legend_frac, box.height])
+        ax.legend(loc="center left", bbox_to_anchor=(1, 0.5))
+        # set x-axis to integers only (since it represents iterations)
+        ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+        ax.set_xlabel("iterations")
+        ax.set_ylabel(yaxis_name)
+        niceplots.adjust_spines(ax)
+        niceplots.save_figs(
+            fig,
+            plot_name,
+            ["png", "svg"],
+            format_kwargs={"png": {"dpi": 400}},
+            bbox_inches="tight",
+        )
+        plt.close("all")
```

Comment:
This is totally a personal opinion, but when I write functions that generate plots, I usually avoid doing anything like saving or closing the figure. This way the same function will still be usable for people who don't want to save the figure as an image (maybe they just want to view it interactively) or want to do some additional formatting before they save it. This is why none of the functions in Niceplots save images, they just return the fig and ax to the user for them to do whatever they want with.

Reply:
I removed this part.